federation/cmd/federation-apiserver/app/server.go

```diff
@@ -196,6 +197,8 @@ func Run(s *options.ServerRunOptions) error {
 		return err
 	}
 
+	routes.Version{}.Install(m.Mux, m.HandlerContainer)
```

Comment:
This is just a drive by since I saw this in my email, but making the
generic apiserver take interfaces to reduce the import dependency tree
sounds great to me :)

On Tue, Sep 13, 2016 at 1:33 PM, Dr. Stefan Schimanski <
notifications@github.com> wrote:

> In federation/cmd/federation-apiserver/app/server.go
> <https://github.com/kubernetes/kubernetes/pull/32387#discussion_r78638591>
> :
>
> > @@ -196,6 +197,8 @@ func Run(s *options.ServerRunOptions) error {
> >  		return err
> >  	}
> >
> > +	routes.Version{}.Install(m.Mux, m.HandlerContainer)
>
> It's kube specific at least as it depends on the Kubernetes version
> package. We could move the whole handler into the version package. How does
> that sound? Another alternative would be to abstract the version getter
> into an interface and inject that via the config.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/32387/files/bd264d43d1c8c0aee12257bcee14754c7a69e761#r78638591>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAngljfVFh7YD0Vn3mZdqUOIkFI4MCG0ks5qpwipgaJpZM4J5JSU>
> .
>